collect_app/src/main/java/org/odk/collect/android/activities/GeoPolyActivity.java

```diff
@@ -49,52 +48,57 @@
+public class GeoPolyActivity extends BaseGeoMapActivity implements IRegisterReceiver {
     public static final String PREF_VALUE_GOOGLE_MAPS = "google_maps";
+    public static final String ANSWER_KEY = "answer";
+    public static final String OUTPUT_MODE_KEY = "output_mode";
     public static final String MAP_CENTER_KEY = "map_center";
     public static final String MAP_ZOOM_KEY = "map_zoom";
     public static final String POINTS_KEY = "points";
-    public static final String RECORDING_ACTIVE_KEY = "recording_active";
-    public static final String RECORDING_MODE_KEY = "recording_mode";
+    public static final String INPUT_ACTIVE_KEY = "input_active";
+    public static final String RECORDING_ENABLED_KEY = "recording_enabled";
+    public static final String RECORDING_AUTOMATIC_KEY = "recording_automatic";
     public static final String INTERVAL_INDEX_KEY = "interval_index";
     public static final String ACCURACY_THRESHOLD_INDEX_KEY = "accuracy_threshold_index";
 
+    public enum OutputMode { GEOTRACE, GEOSHAPE }
+
     private final ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
     private ScheduledFuture schedulerHandler;
 
+    private OutputMode outputMode;
     private MapFragment map;
     private int featureId = -1;  // will be a positive featureId once map is ready
-    private String originalTraceString = "";
+    private String originalAnswerString = "";
 
     private ImageButton zoomButton;
     private ImageButton playButton;
     private ImageButton clearButton;
-    private Button manualButton;
+    private Button recordButton;
     private ImageButton pauseButton;
     private ImageButton backspaceButton;
 
     private TextView locationStatus;
     private TextView collectionStatus;
-    private View traceSettingsView;
-    private View polygonOrPolylineView;
 
-    private AlertDialog traceSettingsDialog;
+    private AlertDialog settingsDialog;
+    private View settingsView;
     private AlertDialog polygonOrPolylineDialog;
+    private View polygonOrPolylineView;
 
     private static final int[] INTERVAL_OPTIONS = {
         1, 5, 10, 20, 30, 60, 300, 600, 1200, 1800
     };
     private static final int DEFAULT_INTERVAL_INDEX = 3; // default is 20 seconds
-    private static final int MANUAL_RECORDING = 0;
-    private static final int AUTOMATIC_RECORDING = 1;
 
     private static final int[] ACCURACY_THRESHOLD_OPTIONS = {
         0, 3, 5, 10, 15, 20
     };
     private static final int DEFAULT_ACCURACY_THRESHOLD_INDEX = 3; // default is 10 meters
 
-    private boolean recordingActive;
-    private int recordingMode; // 0 manual, 1 is automatic
+    private boolean inputActive; // whether we are ready for the user to add points
+    private boolean recordingEnabled; // whether points are taken from GPS readings (if not, placed by tapping)
+    private boolean recordingAutomatic; // whether GPS readings are taken at regular intervals (if not, only when user-directed)
```

Comment:
Nit: Comments like this don't usually get parsed by IDEs for inline advice; if you use `/** JavaDoc comments */` above the var they'll be available elsewhere. Less important here because they're private but still useful.